evm/trie/trie_test.go

```diff
@@ -0,0 +1,365 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package trie
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethdb/leveldb"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+var (
+	defaultTriesToStore = 3
+)
+
+func createNewTxTries(numHistoricalTries int) *TxTries {
+	return NewTxTries(numHistoricalTries)
+}
+
+func createTempDB() *leveldb.Database {
+
+	diskdb, err := leveldb.New("./temp-database", 256, 0, "")
+	if err != nil {
+		panic(fmt.Sprintf("unable to create testing database: %v", err))
+	}
+	return diskdb
+}
+
+func deleteTempDB() {
+	os.RemoveAll("./temp-database")
+}
+
+func addTrie(txTries *TxTries, root common.Hash, transactions []common.Hash, db *leveldb.Database) error {
+
+	if db == nil {
+		db = createTempDB()
+	}
+
+	if transactions == nil {
+		transactions = make([]common.Hash, 0)
+	}
+
+	_, err := txTries.AddTrie(root, db, transactions)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+func TestEmptyTxTries(t *testing.T) {
+	txTries := createNewTxTries(defaultTriesToStore)
+
+	if txTries.triesToStore != defaultTriesToStore {
+		t.Fatalf("tries to store not set properly, expected: %x, got: %x", defaultTriesToStore, txTries.triesToStore)
+	}
+}
+
+func TestAddEmptyTrie(t *testing.T) {
+	txTries := createNewTxTries(defaultTriesToStore)
+	addTrie(txTries, emptyRoot, nil, nil)
```

Comment:
You never check the error that is returned here. You could pass in `t` to the function if you want to avoid having to check everywhere